src/components/base/RegisterForm.tsx

```diff
@@ -0,0 +1,118 @@
+import InputField from './Input'
+import Button from './Button'
+import { useState } from 'react'
+import { Credentials } from '../../entities/Credentials'
+
+export type InputFieldProps = {
+    username?: string
+    password?: string
+
+    repeatPassword?: string
+    name?: string
+    firstname?: string
+    email?: string
+    telephone?: string
+    street?: string
+    place?: string
+    zip?: string
+}
+
+const RegisterForm = (_props: InputFieldProps) => {
+    const [credentials, setCredentials] = useState<Credentials>({
+        username: '',
+        name: '',
+        firstname: '',
+        email: '',
+        telephone: '',
+        street: '',
+        place: '',
+        zip: '',
+        password: '',
+        repeatPassword: '',
+    })
+    const classes = 'ml-13 mr-13 grid gap-x-6 gap-y-2 sm:grid-cols-1 md:grid-cols-2'
+
+    return (
+        <div className="grid gap-4">
+            <div className={classes}>
+                <div>
+                    <InputField
+                        name="Name"
+                        placeholder="Name"
+                        value={credentials.name}
+                        onChange={name => setCredentials({ ...credentials, name })}
```

Comment:
This is the way to do it 👍🏼 When you have the new `onChange` function in the props, you could call here `props.onChange` instead of `setCredentials`.

Reply:
now calling the props.onChange instead of setCredentials